### PR TITLE
Adds the DOTA suit and various extras

### PR DESCRIPTION
**Pull request by BurgerLUA**
## About The Pull Request

Adds the DOTA suit. A special armored science lathe printable hardsuit that gives you morphine whenever you are affected by a wound. Also does other stuff I guess, such as:
- Morphine
- More Morphine
- Emag into giving stimulants (and Morphine)
- Has a cool professionally voiced AI
- Prevents you from talking like a true silent protagonist.
- Built in Geiger counter.

Also adds

- Nerd glasses that give you the "Nice Shot" trait, making your firearms more accurate. How else are you supposed to explain the military training by HEV suit users?
- Fast attacking lightweight crowbar. Beat the shit out of objects. Or your coworkers.
- Physics Manipulation Tool from another PR. I should probably require it to use a bluespace core maybe.

All this is researchable with the correct research.

## Why It's Good For The Game

idk

## Proof Of Testing

image

image

image

## Changelog

:cl: BurgerBB
add: Adds the DOTA suit and various extras to research.
/:cl:

### CODE REVIEW

**Comment by AllCorruptingWyrm** [comment]
i'm not even mad this is fun

**Comment by Yanziko** [comment]
if it works the same as the HEV suit in the gateway with its healing, this should almost certainly be DEEP in research and behind illegal and alien tech, lest ye wish for powergamers to abuse it as hard as they can.

**Comment by projectkepler-ru** [comment]
> if it works the same as the HEV suit in the gateway with its healing, this should almost certainly be DEEP in research and behind illegal and alien tech, lest ye wish for powergamers to abuse it as hard as they can.

it only gives morphine

File: modular_zubbers/code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -66,3 +66,37 @@
 	design_ids -= "mag_autorifle_ic"
 	design_ids += "wt550_ammo_incendiary"
 	. = ..()
+
+/datum/techweb_node/nerd
+	id = "nerd"
+	display_name = "Theoretical Physics"
+	description = "They asked me how well I understood theoretical physics. I said I had a theoretical degree in physics."
+	prereq_ids = list(
```
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/research/techweb/all_nodes.dm:74]: I think basic weapon tech should be added to this aswell

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]

**Comment by OceanFish1** [comment]
oh he's bringing ss13_construct on the station, no way

File: modular_zubbers/code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (BurgerLUA) [reply modular_zubbers/code/modules/research/techweb/all_nodes.dm:74]: "gravity_gun" requires "adv_weaponry" already

**Comment by Yanziko** [comment]
> it only gives morphine

big if true (and also useless) 
in that case, epic

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		addtimer(CALLBACK(src, PROC_REF(process_sound_queue)), 1 SECONDS)
+
+	return TRUE
+
+//Signal handling.
+/obj/item/clothing/suit/armor/nerd/equipped(mob/M, slot)
+	. = ..()
+	if(slot == ITEM_SLOT_OCLOTHING && iscarbon(M))
+		for(var/k in funny_signals)
+			RegisterSignal(M, k, funny_signals[k])
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:124]: Fix up the single letter vars

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+		SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
+		add_queue('modular_zubbers/sound/voice/nerdsuit/morphine.ogg',2 SECONDS)
+
+	COOLDOWN_START(src, next_morphine, MORPHINE_INJECTION_DELAY)
+
+	return TRUE
+
+//General Damage
+/obj/item/clothing/suit/armor/nerd/proc/handle_damage(mob/living/carbon/C, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:220]: See above

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+/obj/item/clothing/suit/armor/nerd/proc/handle_shock(mob/living)
+
+	SIGNAL_HANDLER
+
+	SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
+	add_queue('modular_zubbers/sound/voice/nerdsuit/shock.ogg',3 SECONDS)
+
+//Wounds
+/obj/item/clothing/suit/armor/nerd/proc/handle_wound_add(mob/living/carbon/C, datum/wound/W, obj/item/bodypart/L)
+
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:181]: See above

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	flags_inv = HIDESHOES|HIDEJUMPSUIT|HIDEGLOVES
+
+	armor_type = /datum/armor/nerdsuit
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+
+	strip_delay = 70
+	equip_delay_other = 70
+
+	var/static/list/funny_signals = list(
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:29]: Please rename this.

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+	. = ..()
+	if(slot == ITEM_SLOT_OCLOTHING && iscarbon(M))
+		for(var/k in funny_signals)
+			RegisterSignal(M, k, funny_signals[k])
+		add_queue('modular_zubbers/sound/voice/nerdsuit/bell.ogg',2 SECONDS,purge_queue=TRUE)
+		owner = M
+		if(prob(1))
+			add_queue('modular_zubbers/sound/voice/nerdsuit/emag.ogg',27 SECONDS)
+		else
+			add_queue('modular_zubbers/sound/voice/nerdsuit/welcome.ogg',8 SECONDS)
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:130]:
```suggestion
		add_queue('modular_zubbers/sound/voice/nerdsuit/bell.ogg', 2 SECONDS, purge_queue=TRUE)
		owner = M
		if(prob(1))
			add_queue('modular_zubbers/sound/voice/nerdsuit/emag.ogg', 27 SECONDS)
		else
			add_queue('modular_zubbers/sound/voice/nerdsuit/welcome.ogg', 8 SECONDS)
```

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+/obj/item/clothing/suit/armor/nerd/proc/administer_morphine()
+
+	SIGNAL_HANDLER
+
+	if(!owner.reagents)
+		return
+
+	if(!COOLDOWN_FINISHED(src, next_morphine))
+		return
+
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:203]:
```suggestion
/obj/item/clothing/suit/armor/nerd/proc/administer_morphine()
	SIGNAL_HANDLER

	if(!owner.reagents)
		return
	if(!COOLDOWN_FINISHED(src, next_morphine))
		return

```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Just minor formatting, rest of the code seems understandable and okay, I'll testmerge it in the meantime as to not hold this back

File: modular_zubbers/code/modules/clothing/suits/nerd.dm [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+			COOLDOWN_START(src, next_damage_notify, 5 SECONDS)
+		if(0.26 to 0.50)
+			SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
+			add_queue('modular_zubbers/sound/voice/nerdsuit/vital_signs_critical.ogg',3 SECONDS)
+			COOLDOWN_START(src, next_damage_notify, 5 SECONDS)
+		else
+			SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
+			add_queue('modular_zubbers/sound/voice/nerdsuit/vital_signs_death.ogg',3 SECONDS)
+			COOLDOWN_START(src, next_damage_notify, 5 SECONDS)
+			administer_morphine()
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/clothing/suits/nerd.dm:247]: Can't it be done like this to save on copypasting?

```suggestion
		if(0.51 to 0.75)
			SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_2.ogg')
			add_queue('modular_zubbers/sound/voice/nerdsuit/vital_signs_dropping.ogg',2 SECONDS)
		if(0.26 to 0.50)
			SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
			add_queue('modular_zubbers/sound/voice/nerdsuit/vital_signs_critical.ogg',3 SECONDS)
		else
			SOUND_BEEP('modular_zubbers/sound/voice/nerdsuit/beep_3.ogg')
			add_queue('modular_zubbers/sound/voice/nerdsuit/vital_signs_death.ogg',3 SECONDS)
			administer_morphine()
	COOLDOWN_START(src, next_damage_notify, 5 SECONDS)
```

**Comment by BurgerLUA** [comment]
requested changes above done as well as some balance tweaks

- research for it is easier (people complained it's too much research for a gimmick, including myself)
- the suit no longer makes you mute (people would just take it off and talk, then put it on. it was annoying and not really roleplay friendly

**Comment by projectkepler-ru** [comment]
need to add icon variation for other type of legs

**Comment by BurgerLUA** [comment]
image

**Comment by HissyNays** [comment]
Where's digitigrade legs sprite :(

**Comment by Alecksohs** [comment]
> Where's digitigrade legs sprite :(

image

**Comment by BurgerLUA** [comment]
Someone send me a digigrade sprite template and I will sprite it

**Comment by BurgerLUA** [comment]
I'm serious about the digigrade sprite template. I need it before I can make a digigrade sprite

**Comment by Majkl-J** [comment]
> I'm serious about the digigrade sprite template. I need it before I can make a digigrade sprite

Consider spawning in 4 lizards and screenshotting them with the game on either x1 or x2 scaling. Ill probably provide those on the weekend, so you can most likely do that faster than me

**Comment by BurgerLUA** [comment]
alternatively I can just find the .dmi file for lizards which will take 2 seconds

**Comment by Swiftfeather** [comment]
Put in the digi sprites and we slap merge.

**Pull request merged**